src/parents.js

```diff
@@ -28,13 +34,44 @@ export function setAll (node, options ) {
  * @param {object} parent
  * @param {object} [options]
  * @param {boolean} [options.force] Allow overwriting
+ * @throws {Error} If the child node is not found in the parent
  */
 export function set (node, parent, { force } = {}) {
 	if (!force && parentMap.has(node)) {
 		// We assume that if the node already has a parent, its subtree will also have parents
 		return false;
 	}
-	parentMap.set(node, parent);
+
+	if (parent === undefined) {
+		parentMap.delete(node);
+		return;
+	}
+
+	if (parent === null) {
+		parentMap.set(node, {parent});
+		return;
+	}
+
+	const parentProps = properties[parent?.type] ?? [];
+
+	// Find the property name and index of the child node in the parent
+	for (const property of parentProps) {
+		const child = parent[property];
+		if (Array.isArray(child)) {
+			// When the child is an array, we also need to figure out index
+			const index = child.indexOf(node);
+			if (index !== -1) {
+				parentMap.set(node, {parent, property, index});
+				return;
+			}
+		}
+		else if (child === node) {
+			parentMap.set(node, {parent, property});
+			return;
+		}
+	}
+
+	throw new Error("Could not find child node in parent");
```

Comment:
```suggestion
	throw new Error("No path to child node from parent");
```

(just a suggestion, as "in parent" seemed a little weird to me)